logos-codegen/src/mir.rs

```diff
@@ -111,23 +111,38 @@ impl TryFrom<Hir> for Mir {
 
                 Ok(Mir::Alternation(alternation))
             }
-            HirKind::Literal(literal) => Ok(Mir::Literal(literal)),
+            HirKind::Literal(literal) => {
+                let s = std::str::from_utf8(&*literal.0).unwrap();
+                let mut chars = s.chars().map(Mir::Literal).peekable();
+                let c = chars.next().expect("a literal cannot be empty");
+                if chars.peek().is_some() {
+                    Ok(Mir::Concat(std::iter::once(c).chain(chars).collect()))
```

Comment:
I suspect this is why my PR differs to #320, and doesn't require changing the class priority (producing multiple `Mir::Literal` for one `HirKind::Literal`).